tests/gtest/avifstreamtest.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+#include "avif/internal.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+//------------------------------------------------------------------------------
+
+TEST(StreamTest, Roundtrip) {
+  // TODO(yguyon): Check return values once pull request #1498 is merged.
+
+  // Write some fields.
+  testutil::AvifRwData rw_data;
+  avifRWStream rw_stream;
+  avifRWStreamStart(&rw_stream, &rw_data);
+  EXPECT_EQ(avifRWStreamOffset(&rw_stream), size_t{0});
+
+  const uint8_t rw_somedata[] = {3, 1, 4};
+  avifRWStreamWrite(&rw_stream, rw_somedata, sizeof(rw_somedata));
+
+  const char rw_somechars[] = "somechars";
+  avifRWStreamWriteChars(&rw_stream, rw_somechars,
+                         std::strlen(rw_somechars) + 1);
+
+  const char rw_box_type[] = "type";
+  avifBoxMarker rw_box_marker =
+      avifRWStreamWriteBox(&rw_stream, rw_box_type, /*contentSize=*/0);
+
+  const uint8_t rw_someu8 = 234;
+  avifRWStreamWriteU8(&rw_stream, rw_someu8);
+
+  const int rw_full_box_version = 7;
+  const uint32_t rw_full_box_flags = 0x555;
+  avifBoxMarker rw_full_box_marker =
+      avifRWStreamWriteFullBox(&rw_stream, rw_box_type, /*contentSize=*/0,
+                               rw_full_box_version, rw_full_box_flags);
+
+  const uint16_t rw_someu16 = 2345;
+  avifRWStreamWriteU16(&rw_stream, rw_someu16);
+
+  avifRWStreamFinishBox(&rw_stream, rw_full_box_marker);
+
+  avifRWStreamFinishBox(&rw_stream, rw_box_marker);
+
+  const uint32_t rw_someu32 = 23456;
```

Comment:
23456 is equal to 0x5BA0, which fits in uint16_t also. It seems better to use a contant that won't fit in uint16_t. Similarly for 234567 and uint64_t in line 57.